Data/ByteString/Lazy.hs

```diff
@@ -534,6 +534,41 @@ compareLength Empty toCmp         = compare 0 toCmp
 compareLength (Chunk c cs) toCmp  = compareLength cs (toCmp - S.length c)
 {-# INLINE compareLength #-}
 
+{-# RULES 
+"ByteString.Lazy compareN/length -> compareLength" [~1] forall t n.
+  compare (length t) n = compareLength t (fromIntegral n)
```

Comment:
I wasn't sure about the stage control, so I did it in a similar way as it was done in Data.Text.

Also, since `compareLength` returns an Int and `length` returns an Int64 we need an additional call to `fromIntegral`. I'm not sure if it would be better to change `compareTo` so that it also returns an `Int64` or if there would be drawbacks from that.

Reply:
That's actually a good point: `Data.ByteString.Lazy` API measures length in `Int64` (e. g., [`length`](http://hackage.haskell.org/package/bytestring-0.11.0.0/docs/Data-ByteString-Lazy.html#v:length)), not in `Int`. Please change to `compareLength :: ByteString -> Int64 -> Ordering`.